components/steplist/stories/template.js

```diff
@@ -0,0 +1,117 @@
+import { html, nothing } from "lit-html";
+import { classMap } from "lit-html/directives/class-map.js";
+import { repeat } from "lit-html/directives/repeat.js";
+import { ifDefined } from "lit-html/directives/if-defined.js";
+
+import { Template as Tooltip } from '@spectrum-css/tooltip/stories/template.js';
+
+import "../index.css";
+import "../skin.css";
+
+export const SteplistItem = ({
+  rootClass,
+  isSmall = false,
+  isInteractive = false,
+  withTooltip = false,
+  label,
+  ariaPosInSet = 1,
+  ariaSetSize = 4,
+  isComplete = false,
+  isSelected = false,
+  id,
+  ...globals
+}) => {
+  const labelMarkup = !isSmall && !withTooltip && typeof label !== "undefined"
+    ? html`<span class="spectrum-Steplist-label">${label}</span>`
+    : nothing;
+
+  const markerContainer = html`
+    <span class="${rootClass}-markerContainer">
+      ${ withTooltip && !isSmall && typeof label !== "undefined"
+          ? Tooltip({
+              label,
+              isOpen: false,
+              placement: 'top',
+              showOnHover: true,
+            })
+          : nothing
+      }
+      <span class="${rootClass}-marker"></span>
+    </span>
+  `;
+
+  return html`
+    <div
+      class=${classMap({
+        [`${rootClass}-item`]: true,
+        "is-complete": isComplete,
+        "is-selected": isSelected,
+        'u-tooltip-showOnHover': withTooltip && !isSmall && typeof label != "undefined",
+      })}
+      id=${ifDefined(id)}
+      role="listitem"
+      aria-posinset=${ariaPosInSet}
+      aria-setsize=${ariaSetSize}
+      aria-label=${isSmall && !isInteractive ? ifDefined(label) : nothing}
```

Comment:
Haven't seen `nothing` before but it works?! Cool!

Reply:
> Haven't seen `nothing` before but it works?! Cool!

Yeah, I just learned about it [here](https://lit.dev/docs/templates/conditionals/#conditionally-rendering-nothing). In attribute values it'll remove the attribute, similar to how `ifDefined` does. But also can render nothing for a whole expression. It looks very useful.